examples/qoi.cpp

```diff
@@ -0,0 +1,251 @@
+          --max-levels 2 --fine-samples 0 \
+          --coarse-samples 5 --shared-samples 5 --choose-samples 90
+
+   Each gives essentially the same estimate for the quantity of interest, but
+   the two-level estimator is much more efficient because most samples are
+   taken on the coarse grid. Also note in the two-level run that the
+   variance for the correction on the fine grid is much smaller than the
+   variance on the coarse grid, which is what justifies taking few samples
+   on the fine grid.
+
+   See Osborn, Vassilevski, and Villa, A multilevel, hierarchical sampling
+   technique for spatially correlated random fields, SISC 39 (2017) pp. S543-S562
+   and its references for more information on these techniques.
+
+   Output from this example can be visualized (in 2D) with examples/qoivis.py
+*/
+
+#include <fstream>
+#include <sstream>
+#include <mpi.h>
+
+#include "../src/picojson.h"
+#include "../src/smoothG.hpp"
+#include "pde.hpp"
+
+using namespace smoothg;
+
+int main(int argc, char* argv[])
+{
+    int num_procs, myid;
+    picojson::object serialize;
+
+    mpi_session session(argc, argv);
+    MPI_Comm comm = MPI_COMM_WORLD;
+    MPI_Comm_size(comm, &num_procs);
+    MPI_Comm_rank(comm, &myid);
+
+    // program options from command line
+    UpscaleParameters upscale_param;
+    mfem::OptionsParser args(argc, argv);
+    double kappa = 0.001;
+    args.AddOption(&kappa, "--kappa", "--kappa",
+                   "(Inverse) correlation length for Gaussian samples.");
+    int dump_number = 3;
+    args.AddOption(&dump_number, "--dump-number", "--dump-number",
+                   "Number of samples for which to dump solutions for eg. visualization.");
+    int shared_samples = 3;
+    args.AddOption(&shared_samples, "--shared-samples", "--shared-samples",
+                   "(minimum) Number of samples to draw and simulate on all levels, "
+                   "estimating difference.");
+    int fine_samples = 0;
+    args.AddOption(&fine_samples, "--fine-samples", "--fine-samples",
+                   "Number of samples to draw and simulate on the fine level alone,"
+                   " normally only for debugging/comparison.");
+    int coarse_samples = 3;
+    args.AddOption(&coarse_samples, "--coarse-samples", "--coarse-samples",
+                   "(minimum) Number of samples to draw and simulate on the coarse level alone.");
+    int choose_samples = 10;
+    args.AddOption(&choose_samples, "--choose-samples", "--choose-samples",
+                   "Number of samples where MLMC will choose to optimize.");
+    int argseed = 1;
+    args.AddOption(&argseed, "--seed", "--seed", "Seed for random number generator.");
+    int dimension = 2;
+    args.AddOption(&dimension, "--dimension", "--dimension",
+                   "Spatial dimension of simulation.");
+    double cell_volume = 10000.0;
+    args.AddOption(&cell_volume, "--volume", "--volume",
+                   "Volume of typical cell (for PDE sampler).");
```

Comment:
Should this be obtained from `DarcyProblem::CellVolume` or `pmesh_`? It should match with `scale`.